pkg/rbd/controllerserver.go

```diff
@@ -156,17 +158,24 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	volumeID := req.GetVolumeId()
 	rbdVol := &rbdVolume{}
 	if err := loadVolInfo(volumeID, path.Join(PluginFolder, "controller"), rbdVol); err != nil {
+		if os.IsNotExist(errors.Cause(err)) {
+			// Must have been deleted already. This is not an error (idempotency!).
+			return &csi.DeleteVolumeResponse{}, nil
+		}
 		return nil, err
 	}
 	volName := rbdVol.VolName
 	// Deleting rbd image
 	glog.V(4).Infof("deleting volume %s", volName)
 	if err := deleteRBDImage(rbdVol, rbdVol.AdminId, req.GetControllerDeleteSecrets()); err != nil {
```

Comment:
one more TODO: hold a mutex here so no two delete can happen in parallel.

Reply:
That's basically the "Serialize?" in the last TODO. But I can also move that up.